spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/blocking/retry/ClientBlockingLoadBalancedRetryFactory.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2012-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.loadbalancer.blocking.retry;
+
+import org.springframework.cloud.client.loadbalancer.ClientLoadBalancerProperties;
+import org.springframework.cloud.client.loadbalancer.LoadBalancedRetryFactory;
+import org.springframework.cloud.client.loadbalancer.LoadBalancedRetryPolicy;
+import org.springframework.cloud.client.loadbalancer.LoadBalancerProperties;
+import org.springframework.cloud.client.loadbalancer.ServiceInstanceChooser;
+import org.springframework.cloud.loadbalancer.blocking.client.BlockingLoadBalancerClient;
+
+/**
+ * An implementation of {@link LoadBalancedRetryFactory} for client specific
+ * {@link BlockingLoadBalancerClient}.
+ *
+ * @author Andrii Bohutskyi
+ */
+public class ClientBlockingLoadBalancedRetryFactory implements LoadBalancedRetryFactory {
+
+	private final LoadBalancerProperties loadBalancerProperties;
+
+	private final ClientLoadBalancerProperties clientLoadBalancerProperties;
+
+	public ClientBlockingLoadBalancedRetryFactory(LoadBalancerProperties loadBalancerProperties,
+			ClientLoadBalancerProperties clientLoadBalancerProperties) {
+		this.loadBalancerProperties = loadBalancerProperties;
+		this.clientLoadBalancerProperties = clientLoadBalancerProperties;
+	}
+
+	@Override
+	public LoadBalancedRetryPolicy createRetryPolicy(String service, ServiceInstanceChooser serviceInstanceChooser) {
+		final LoadBalancerProperties properties = clientLoadBalancerProperties.getClientLoadBalancerProperties(service)
```

Comment:
@Ferioney , as discussed [here](https://github.com/spring-cloud/spring-cloud-commons/issues/914#issuecomment-840422056), we will wait with resolving properties per client in general till the feature is implemented in Boot, so the changes in this class should be removed. Specifcally, the issue here is that either all properties are resolved "per client" for a given client or none, while in the final version, we would want it to be possible to make that distinction on property level. For now, a simpler implementation only for the specific properties can be added to fix this issue, similarly to what is being done [here](https://github.com/spring-cloud/spring-cloud-commons/blob/779d482730dc2c8b988056e594ec664733e95010/spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/HealthCheckServiceInstanceListSupplier.java#L66) and [here](https://github.com/spring-cloud/spring-cloud-commons/blob/779d482730dc2c8b988056e594ec664733e95010/spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/HealthCheckServiceInstanceListSupplier.java#L130-L131).